lib/process/process.go

```diff
@@ -1813,7 +1813,10 @@ func (p *Process) initAccount() error {
 }
 
 // removeLegacyIdentities removes legacy admin/proxy identities so that new
-// ones can be generated new ones upon the first start of new teleport
+// ones can be generated upon the first start of new teleport
+//
+// If they are not removed, this process that includes teleport 3.0 will not
+// be able to start after upgrade from older gravity that used teleport 2.4.
```

Comment:
So, as @knisbet suggested, this should get a TODO to remove once 2.4 upgrades are no longer supported (?)